apps/desktop/desktop_native/core/src/crypto/crypto.rs

```diff
@@ -37,3 +37,33 @@ pub fn encrypt_aes256(
 
     Ok(CipherString::AesCbc256_B64 { iv, data })
 }
+
+pub fn argon2(
+    secret: &[u8],
+    salt: &[u8],
+    iterations: u32,
+    memory: u32,
+    parallelism: u32,
+) -> Result<[u8; 32]> {
+    use argon2::*;
+
+    let params = Params::new(memory, iterations, parallelism, Some(32)).map_err(|e| {
+        KdfParamError::InvalidParams(format!(
+            "Argon2 parameters are invalid: {}",
+            e.to_string()
```

Comment:
Tiny nit, this can be simplified slightly like this, which uses the error Display impl:
```suggestion
            "Argon2 parameters are invalid: {e}"
```

Reply:
Updated, nice!